blivet/devicefactory.py

```diff
@@ -1482,21 +1479,15 @@ def _get_total_space(self):
                 size -= self.pool.free_space
                 log.debug("size cut to %s to omit pool free space", size)
 
-                pad = Size(blockdev.lvm.get_thpool_padding(self.pool.free_space, self._pesize))
-                size -= pad
-                log.debug("size cut to %s to omit pool padding from free "
-                          "space", size)
+        if self.container:
+            self.container.thpool_reserve = AUTOPART_THPOOL_RESERVE
 
-            if self.device and self.container_size == SIZE_POLICY_AUTO:
-                # Now we have to reduce the space again by the current device's
-                # portion of the pool padding.
-                # The member count here uses the container's current member set
-                # since that's the basis for the current device's disk space
-                # usage.
-                pad = Size(blockdev.lvm.get_thpool_padding(self.device.size, self._pesize))
-                log.debug("old device size: %s ; old pad: %s", self.device.size, pad)
-                size -= pad
-                log.debug("size cut to %s to omit old device padding", size)
+        # black maths (to make sure there's AUTOPART_THPOOL_RESERVE.percent reserve)
+        size_with_reserve = size * (1 / (1 - (AUTOPART_THPOOL_RESERVE.percent / 100)))
+        if size_with_reserve - size > AUTOPART_THPOOL_RESERVE.min:
+            size = size_with_reserve
+        else:
+            size = size + AUTOPART_THPOOL_RESERVE.min
```

Comment:
You mean the logic of ``thpool_reserve``? Ideally it should be the VG saying *"I want this much more"*, but that would be (is going to be) quite a big change of logic which I'm trying to avoid here. So this just instead tries to be clever enough to give the VG enough space for it to have it's reserved space.

And now that I see it, it should also reflect ``AUTOPART_THPOOL_RESERVE.max`` because it may quite easily happen that somebody will use device factory for a VG bigger than 500 GiB.